services/spar/test-integration/Test/Spar/SCIMSpec.hs

```diff
@@ -38,7 +39,7 @@ import qualified Data.Aeson as Aeson
 spec :: SpecWith TestEnv
 spec = do
     specUsers
-
+    specTokens
```

Comment:
missing tests (i think; correct me if i am wrong):

- if idp is not provided, the create-token request fails (or is it the SAML login request later on that fails?)
- come to think of it, test that SCIM and saml work together and SCIM-created users can login.  give this task back to me if you like, I've recently written a lot of test code that can be re-used for this.
- `Data.getScimTokens` is not tested here.  I guess you may as well add an end-point for that now in `APIScimToken`

(I personally prefer to test the database interface separately, but I don't want to insist.)

Reply:
@fisx 

> give this task back to me if you like, I've recently written a lot of test code that can be re-used for this.

Would you like to do it in a new PR, or in this one?

Reply:
> if idp is not provided, the create-token request fails (or is it the SAML login request later on that fails?)

You don't provide the IdP when creating the token; the backend looks at the list of IdPs for your team and fails unless there's exactly one. I can add a test like "token can't be created for an IdP-less team", though.

Reply:
> @fisx
> 
> > give this task back to me if you like, I've recently written a lot of test code that can be re-used for this.
> 
> Would you like to do it in a new PR, or in this one?

ok, fine, i'll do it somewhere else.  :-)  consider it solved!

Reply:
> Data.getScimTokens is not tested here. I guess you may as well add an end-point for that now in APIScimToken

I'll do that and test it.

> I personally prefer to test the database interface separately.

Can you very briefly describe what kind of database tests you'd like to see? Like, in pseudocode.